### ISSUE TITLE
Intermittent Playback OMX.brcm.video.hw.Decoder Error on Amazon FireTV with DASH/Playready content

### ISSUE THREAD

**Issue by rs146** [google/ExoPlayer#7892]
### [REQUIRED] Searched documentation and issues
I have had a look on the Issue tracker and StackOverflow - but to no avail.

### [REQUIRED] Question
We are finding that we get the following error intermittently on our production application on 1st generation Amazon fire sticks. 
```
MediaCodecRenderer: Failed to initialize decoder: OMX.brcm.video.hw.decoder.secure
      java.lang.IllegalArgumentException: Failed to initialize OMX.brcm.video.hw.decoder.secure, error 0xfffffffe
```

Please note - I cannot reproduce this error on the ExoPlayer demo application (that's why this has been raised as a question). 

I can confirm our release logic (`player`, `adsLoader` and the `trackSelector`) matches the ExoPlayer demo application. We would be grateful if you could help understand why this decoder initialisation error occurs and how we could rectify this given that it does not appear in the ExoPlayer demo application.

We are using ExoPlayer version 2.11.8.
Playback: DASH
DRM: Playready

### A full bug report captured from the device
Unfortunately as this is an older Android device - I can only provide a Logcat of the issue. The first part of the Logcat shows the decoder being initialised successfully for successful playback startup. The second part of the Logcat shows the stacktrace when the user exists the first playback and attempts to start the same DASH manifest.

[fire-stick-gen-1-playready.txt](https://github.com/google/ExoPlayer/files/5210103/fire-stick-gen-1-playready.txt)

**Comment by kim-vde** [comment]
`omxError 0x80001003` means that no component with the specified name was found.

Are you using the same player instance when playing the stream for the second time? If not, do you release the first player before using the second one? It could be that the codec is still locked by the first player while the second player is trying to get it.

Do you have the same issue with another codec? You can try to exclude this codec by configuring a `MediaCodecSelector` in the `DefaultRenderersFactory`.

**Comment by rs146** [comment]
Thank you for your reply - this is very helpful. I can confirm that we do not use the same player instance when playing the stream for the second time. I will have another look at the code - we do release the player before we start playback and this could be causing it.

**Comment by kim-vde** [comment]
I will close this issue as there hasn't been any activity for 14 days. Feel free to open a new issue if you still have problems.
